server/service/src/sync/integrate_document.rs

```diff
@@ -16,8 +16,18 @@ use crate::{
         program_enrolment::program_schema::SchemaProgramEnrolment,
     },
 };
+#[derive(Debug)]
+pub(crate) struct DocumentUpsert(pub(crate) Document);
 
-pub(crate) fn sync_upsert_document(
+impl Upsert for DocumentUpsert {
+    fn upsert_sync(&self, con: &StorageConnection) -> Result<(), RepositoryError> {
+        sync_upsert_document(con, &self.0)
+    }
+
+    fn assert_upserted(&self, _: &StorageConnection) {}
```

Comment:
Thanks !

I guess this would be a good use case of splitting UpsertTest and Upsert ? (UpsertTest can go into repo), although this kind of separates the test and upsert logic, so not sure:

> Shouldn't this also go into repository/src/db_diesel/document.rs?

I wasn't sure how to split this, since UpsertDocument does quite a lot of things, not just inserting document, would your suggestion above be enough for now ?